repository_test.go

```diff
@@ -1071,6 +1071,49 @@ func (s *RepositorySuite) testCloneSingleBranchAndNonHEADReference(c *C, ref str
 	c.Assert(branch.Hash().String(), Equals, "e8d3ffab552895c19b9fcf7aa264d277cde33881")
 }
 
+func (s *RepositorySuite) TestCloneSingleBranchHEADMain(c *C) {
+	r, _ := Init(memory.NewStorage(), nil)
+
+	head, err := r.Head()
+	c.Assert(err, Equals, plumbing.ErrReferenceNotFound)
+	c.Assert(head, IsNil)
+
+	err = r.clone(context.Background(), &CloneOptions{
+		URL:          "https://github.com/AriehSchneier/main.git",
```

Comment:
@pjbgf I would appreciate if you created me a repo in [`git-fixtures`](https://github.com/git-fixtures) to use instead

Reply:
@mcuadros do you mind creating a new repository where default branch is `main` please?

Reply:
I found a fixture that has a `main` and no `master`. I forced pushed an update, the only difference is the clone URL is now from fixtures and the hash check is updated.

Reply:
@pjbgf I found a fixture that has a main and no master. I forced pushed an update, the only difference is the clone URL is now from fixtures and the hash check is updated.